src/request_blocks.cpp

```diff
@@ -83,8 +83,7 @@ namespace libtorrent {
 
 		// in time critical mode, only have 1 outstanding request at a time
 		// via normal requests
-		int const desired_queue_size = time_critical_mode
-			? 1 : c.desired_queue_size();
+		int const desired_queue_size = c.desired_queue_size();
```

Comment:
does it still work if you make this use `c.desired_queue_size() / 2` in `time_critical_mode`? If so, I think that would be a bit safer